sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobStorageTesting/StorageScope.cs

```diff
@@ -100,63 +87,37 @@ public static async Task<StorageScope> CreateAsync([CallerMemberName] string cal
-                return new StorageScope(container.Name);
-            }
-        }
+            var attempts = 0;
+            var capturedException = default(Exception);
 
-        /// <summary>
-        ///   Performs the tasks needed to create a new Azure storage account within a resource group, intended to be used as
-        ///   an ephemeral container for the Event Hub instances used in a given test run.
-        /// </summary>
-        ///
-        /// <returns>The key attributes for identifying and accessing a dynamically created Azure storage account.</returns>
-        ///
-        public static async Task<StorageTestEnvironment.StorageProperties> CreateStorageAccountAsync()
-        {
-            var subscription = EventHubsTestEnvironment.Instance.SubscriptionId;
-            var resourceGroup = EventHubsTestEnvironment.Instance.ResourceGroup;
-            var token = await ResourceManager.AcquireManagementTokenAsync().ConfigureAwait(false);
-
-            static string CreateName() => $"neteventhubs{ Guid.NewGuid().ToString("N").Substring(0, 12) }";
+            // There is an unlikely possibility of a name collision for tests with longer names; allow retrying
+            // with a new name after the client has applied its default policy.
 
-            using (var client = new StorageManagementClient(AzureResourceManagerUri, new TokenCredentials(token)) { SubscriptionId = subscription })
+            while (++attempts < LiveResourceManager.RetryMaximumAttempts)
             {
-                var location = await ResourceManager.QueryResourceGroupLocationAsync(token, resourceGroup, subscription).ConfigureAwait(false);
-                var sku = new Sku(SkuName.StandardLRS, SkuTier.Standard);
-                var parameters = new StorageAccountCreateParameters(sku, Kind.BlobStorage, location: location, tags: ResourceManager.GenerateTags(), accessTier: AccessTier.Hot);
-                StorageAccount storageAccount = await ResourceManager.CreateRetryPolicy<StorageAccount>().ExecuteAsync(() => client.StorageAccounts.CreateAsync(resourceGroup, CreateName(), parameters)).ConfigureAwait(false);
-
-                StorageAccountListKeysResult storageKeys = await ResourceManager.CreateRetryPolicy<StorageAccountListKeysResult>().ExecuteAsync(() => client.StorageAccounts.ListKeysAsync(resourceGroup, storageAccount.Name)).ConfigureAwait(false);
-                return new StorageTestEnvironment.StorageProperties(storageAccount.Name, $"DefaultEndpointsProtocol=https;AccountName={ storageAccount.Name };AccountKey={ storageKeys.Keys[0].Value };EndpointSuffix={ StorageTestEnvironment.Instance.StorageEndpointSuffix }", shouldRemoveAtCompletion: true);
+                try
+                {
+                    var containerClient = new BlobContainerClient(StorageTestEnvironment.Instance.StorageConnectionString, CreateName());
```

Comment:
For my own learning: Is it necessary to create a new client per attempt?

Reply:
Under normal circumstances, no.   

I'm doing so here only because we're generating the container name using a partially random pattern;  we've seen a couple of occurrences where we've had a name collision, either because the table was created during an earlier attempt and we timed out before receiving the response or a collision in the name generation.   This approach is used just to ensure a clean state for the test and avoid more complex failure handling (like scanning the container and trying to determine if it is/was in use).